### PR TITLE
PP-736: update log_match in PTL to report failure

### PR DESCRIPTION
**Pull request by hirenvadalia**
#### Issue-ID

* **[PP-736](https://pbspro.atlassian.net/browse/PP-736)**

#### Problem description
* Make log_match method to fail when no match found for given log message so that user does not have to assert in test case

#### Cause / Analysis
* Currently log_match returns None when no match found for given log message and so test writer has to capture that None and based on that return value test writer has to write assert call

#### Solution description
* modified log_match method to throw assert failure whenever no match found for given log message

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
- [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [ ] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
- [ ] I have added  **manual test(s) to the Jira ticket and explained why PTL is not appropriate** for this case.
- [x] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [x] I have attached **test logs to the Jira ticket** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

**Comment by bhroam** [comment]
I don't see any changes to the log_match() calls in pbs_smoketest.py.  

I find the try/except block a bit cumbersome to check for the absence of a log message.  Could you offer an argument to log_match() that says if you're looking for the log message to exist or not?  It can be True by default and can be set to False if you are looking for the absence of a log message.

**Comment by mike0042** [comment]
You could also add a new log_missing() message that calls log_match() with the switch @bhroam suggested. That might make the code a bit more readable.

**Comment by bhroam** [comment]
What is the status of this PR?  It's been lingering for quite some time.

**Comment by hirenvadalia** [comment]
@bhroam As per your suggestion I have added new argument to *.log_match, Server.accounting_match and Server.tracejob_match called 'existence'. Also pre and post fix test results are added to ticket.
Please have look and let me know if anything is missing.

File: test/tests/functional/pbs_Rrecord_resources_used.py [hunk 1]
```diff
@@ -378,26 +351,21 @@ def test_t5(self):
...
+        self.server.accounting_match(
             msg='.*Resource_List.*', id=jid1, regexp=True)
-        self.assertNotEqual(m, None)
-        m = self.server.accounting_match(
-            msg='.*R;' + jid1 + '.*resources_used.*', id=jid1, regexp=True)
-        self.assertIsNone(m)
-        m = self.server.accounting_match(
+        msg = '.*R;' + jid1 + '.*resources_used.*'
+        self.server.accounting_match(msg, id=jid1, regexp=True,
+                                     existence=False)
```
Comment (bhroam) [inline test/tests/functional/pbs_Rrecord_resources_used.py]: In the previous call to accounting_match() and other calls in this file, you call accounting_match(msg=X).  In this case the message has been pulled out and set into a variable which is passed into accounting_match() as the first parameter.  Why the difference?

Comment (hirenvadalia) [inline test/tests/functional/pbs_Rrecord_resources_used.py]: @bhroam Thanks for review. There is no difference in call. I just formatted little bit. Also I removed 'msg=' from call as first argument to accounting_match is message and I am still passing that using msg variable.

Comment (hirenvadalia) [inline test/tests/functional/pbs_Rrecord_resources_used.py]: @bhroam Any update on this?

Comment (bhroam) [inline test/tests/functional/pbs_Rrecord_resources_used.py]: Oh sorry, I was waiting for a new commit.  I thought you were saying that you were changing something and were about to commit something.  If that isn't the case then I'd still like to see you be consistent in the use of either passing the first argument as the msg variable or msg=X argument.

**Comment by hirenvadalia** [comment]
@bhroam I have update code as per your comment. Please have a look and let me know if anything is missing...

**Comment by bhroam** [comment]
Looks good to me.  I sign off.  Please squash and rebase for the final merge.

**Comment by hirenvadalia** [comment]
@bhroam and @kjakkali Thanks for review. @bhroam I have rebased branch with current master and squashed all commits. Please take it further.

**Comment by bhroam** [comment]
I merged in the PR.  We now should probably catch all the current PRs in progress using log_match() to no longer check for the return value.

**Comment by hirenvadalia** [comment]
@bhroam Thanks. And regarding all current PR I will try my best to catch all PR who is using log_match(). But extra hands are always welcome 😄

**Pull request merged**